socorro/processor/rules/breakpad.py

```diff
@@ -200,23 +200,37 @@ def action(self, raw_crash, dumps, processed_crash, tmpdir, status):
                     self.truncate_stacks(value["json_dump"], f"{key}.json_dump", status)
 
 
-def execute_process(command_line):
+def execute_process(command_line, timeout=120):
     """Executes process and returns completed process data.
 
     :param command_line: the complete command line to run
+    :param timeout: the timeout in seconds for the command to complete before it's
+        killed; defaults to 120 seconds
 
-    :returns: dict with stdout, stderr, and returncode keys
+    :returns: dict with stdout (bytes), stderr (bytes), and returncode (signed smallint)
+        keys
 
     """
     # Tokenize the command line into args
     args = shlex.split(command_line, comments=False, posix=True)
-    completed = subprocess.run(args, capture_output=True)
-    ret = {
-        # NOTE(willkg) stdout and stderr here are bytes
-        "stdout": completed.stdout,
-        "stderr": completed.stderr,
-        "returncode": completed.returncode,
-    }
+    try:
+        completed = subprocess.run(args, timeout=timeout, capture_output=True)
+        ret = {
+            # NOTE(willkg) stdout and stderr here are bytes
+            "stdout": completed.stdout,
+            "stderr": completed.stderr,
+            "returncode": completed.returncode,
+        }
+
+    except subprocess.TimeoutExpired as timeout_exc:
+        ret = {
+            # NOTE(willkg) stdout and stderr here are bytes
+            "stdout": timeout_exc.stdout or b"",
```

Comment:
`stdout` is an alias for `output` and it can be `None` in the case where there's no output. This normalizes things so we're always getting at least an empty byte string.